src/distance/ConfigurationRanker.cpp

```diff
@@ -0,0 +1,81 @@
+#include "aikido/distance/ConfigurationRanker.hpp"
+
+namespace {
+
+struct sortByCost
+{
+  bool operator()(
+      const std::pair<aikido::statespace::CartesianProduct::State*, double>&
+          left,
+      const std::pair<aikido::statespace::CartesianProduct::State*, double>&
+          right)
+  {
+    return left.second < right.second;
+  }
+};
+}
+
+namespace aikido {
+namespace distance {
+
+using statespace::dart::ConstMetaSkeletonStateSpacePtr;
+using ::dart::dynamics::ConstMetaSkeletonPtr;
+
+//==============================================================================
+ConfigurationRanker::ConfigurationRanker(
+    ConstMetaSkeletonStateSpacePtr metaSkeletonStateSpace,
+    ConstMetaSkeletonPtr metaSkeleton)
+  : mMetaSkeletonStateSpace(std::move(metaSkeletonStateSpace))
+  , mMetaSkeleton(std::move(metaSkeleton))
+{
+  if (!mMetaSkeletonStateSpace)
+    throw std::invalid_argument("MetaSkeletonStateSpace is nullptr.");
+
+  if (!mMetaSkeleton)
+    throw std::invalid_argument("MetaSkeleton is nullptr.");
+
+  mDistanceMetric = createDistanceMetricFor(
+      std::dynamic_pointer_cast<statespace::CartesianProduct>(
```

Comment:
Are both of these casts necessary? Does `createDistanceMetricFor` not work for `const statespace::CartesianProduct`?

Reply:
The const-cast should not be necessary.